docs/docs/validating-plugin-options.md

```diff
@@ -0,0 +1,224 @@
+
+Make sure that every option and field has a `.description()` explaining its purpose. This is helpful for documentation as users can look at the schema and understand all the options. There might also be tooling in the future that auto-generates plugin option documentation from the schema.
+
+### Set default options
+
+You can use the `.default()` method to set a default value for an option. For example with `gatsby-plugin-console-log`, you can have the `message` option default to `"default message"` in case a user does not pass their own `message`:
+
+```javascript:title=plugins/gatsby-plugin-console-log/gatsby-node.js
+exports.pluginOptionsSchema = ({ Joi }) => {
+  return Joi.object({
+    optionA: Joi.boolean().required().description(`Enables optionA.`),
+    message: Joi.string()
+      .default(`default message`) // highlight-line
+      .description(`The message logged to the console.`),
+    optionB: Joi.boolean().description(`Enables optionB.`),
+  })
+}
+```
+
+Accessing `pluginOptions.message` would then log `"default message"` in all plugin APIs if the user does not supply their own value:
+
+```javascript:title=plugins/gatsby-plugin-console-log/gatsby-node.js
+exports.onPreInit = (_, pluginOptions) => {
+  console.log(
+    `logging: "${pluginOptions.message}" to the console` // highlight-line
+  )
+}
+```
+
+### Validating external access
+
+Some plugins (particularly source plugins) query external APIs. With the `.external()` method, you can asynchronously validate that the user has access to the API, providing a better experience if they pass invalid secrets.
+
+For example, this is how the [Contentful source plugin](/plugins/gatsby-source-contentful/) ensures the user has access to the space they are trying to query:
+
+```javascript:title=gatsby-source-contentful/gatsby-node.js
+exports.pluginOptionsSchema = ({ Joi }) => {
+  return Joi.object({
+    accessToken: Joi.string().required(),
+    spaceId: Joi.string().required(),
+    // ...more options here...
+  }).external(async pluginOptions => {
+    try {
+      await contentful
+        .createClient({
+          space: pluginOptions.spaceId,
+          accessToken: pluginOptions.accessToken,
+        })
+        .getSpace()
+    } catch (err) {
+      throw new Error(
+        `Cannot access Contentful space "${pluginOptions.spaceId}" with the provided access token. Double check they are correct and try again!`
+      )
+    }
+  })
+}
+```
+
+### Add custom error messages
+
+Sometimes you might want to provide more detailed error messages when validation fails for a specific field. Joi provides a [`.messages()` method](https://joi.dev/api/?v=17.2.1#anymessagesmessages) which lets you override error messages for specific [error types](https://joi.dev/api/?v=17.2.1#list-of-errors) (e.g. `"any.required"` when a `.required()` call fails).
+
+For example, this is how we would provide a custom error message if users do not specify `optionA` for `gatsby-plugin-console-log`:
```

Comment:
Per the style guide, [use "you" instead of "we"](https://www.gatsbyjs.com/contributing/gatsby-style-guide/#use-you-as-the-pronoun).

```suggestion
For example, in the `gatsby-plugin-console-log` plugin above, this is how you would provide a custom error message if users do not specify `optionA`:
```